okhttp/src/main/kotlin/okhttp3/OkHttpClient.kt

```diff
@@ -269,14 +269,19 @@ open class OkHttpClient internal constructor(
 
   /** Uses [request] to connect a new web socket. */
   override fun newWebSocket(request: Request, listener: WebSocketListener): WebSocket {
+    return newWebSocket(request, listener, null)
+  }
+
+  fun newWebSocket(request: Request, listener: WebSocketListener, pingPayloadFactory: WebSocketPingPayloadFactory?): WebSocket {
```

Comment:
The change overall seems relatively safe and clean. I think the main concern overall is the new API here and in WebSocketListener.  I wonder whether a subinterface specific to ping-pong handling could be a good way to control this instead.  If the listener implements that we will use it. It would minimise surface area and not expose this as top level API and we could mark that interface as experimental?

@swankjesse @JakeWharton